README.md

```diff
@@ -1,13 +1,192 @@
+
+- Swift 
+```swift
+extension AppDelegate: PurchaseRevenueDataSource, PurchaseRevenueDelegate {
+    // PurchaseRevenueDelegate method implementation
+    func didReceivePurchaseRevenueValidationInfo(_ validationInfo: [AnyHashable : Any]?, error: Error?) {
+        print("PurchaseRevenueDelegate: \(validationInfo)")
+        print("PurchaseRevenueDelegate: \(error)")
+      // process validationInfo here 
+}
+    // PurchaseRevenueDataSource method implementation
+    func purchaseRevenueAdditionalParameters(for products: Set<SKProduct>, transactions: Set<SKPaymentTransaction>?) -> [AnyHashable : Any]? {
+        // Add additional parameters for SKTransactions here.
+        return ["additionalParameters":["param1":"value1", "param2":"value2"]];
+    }
+}
+```
+
+- Objective-C 
+```objective-c
+@interface AppDelegate () <AppsFlyerPurchaseRevenueDelegate, AppsFlyerPurchaseRevenueDataSource>
+@end
+
+@implementation AppDelegate
+
+- (void)applicationDidBecomeActive:(UIApplication *)application {
+    [[PurchaseConnector shared] startObservingTransactions];
+    [[AppsFlyerLib shared] start];
+}
+
+- (void)didReceivePurchaseRevenueValidationInfo:(NSDictionary *)validationInfo error:(NSError *)error {
+    NSLog(@"Validation info: %@", validationInfo);
+    NSLog(@"Error: %@", error);
+    
+    // Process validation info
+}
+
+- (NSDictionary *)purchaseRevenueAdditionalParametersForProducts:(NSSet<SKProduct *> *)products transactions:(NSSet<SKPaymentTransaction *> *)transactions {
+    return @{@"key1" : @"param1"};
+}
+
+@end
+```
+
+### <a id="start"> Start Observing Transactions
+
+`startObservingTransactions` should be called to start observing transactions.
+
+- Swift 
+```swift
+    PurchaseConnector.shared().startObservingTransactions()
+```
+
+- Objective-C 
+```objective-c
+    [[PurchaseConnector shared] startObservingTransactions];
+```
+
+### <a id="stop"> Stop Observing Transactions
+
+To stop observing transactions, you need to call `stopObservingTransactions`.
+
+- Swift 
+```swift
+    PurchaseConnector.shared().stopObservingTransactions()
+```
+
+- Objective-C 
+```objective-c
+    [[PurchaseConnector shared] stopObservingTransactions];
+```
+
+> *Note: if you called `stopObservingTransactions` API, you should set `autoLogPurchaseRevenue` value before you call `startObservingTransactions` next time.*
+
+## <a id="testing"> Testing the implementation in Sandbox
+
+In order to test purchases in Xcode environment on a real device with TestFlight sandbox account, you need to set `isSandbox` to true.
+
+- Swift 
+```swift
+    PurchaseConnector.shared().isSandbox = true
+```
+
+- Objective-C 
+```objective-c
+    [[PurchaseConnector shared] setIsSandbox:YES];
+```
+
+> *IMPORTANT NOTE: Before releasing your app to production, please be sure, to remove `isSandbox` or set it to `false`. If the production purchase event will be sent in sandbox mode, your event will not be validated propperly! *
```

Comment:
- please be sure to remove (without comma)
- "properly" instead of "propperly" (remove extra p)